lang/cs/Org.Apache.REEF.Wake/Remote/Impl/StreamingLink.cs

```diff
@@ -167,20 +162,12 @@ public T Read()
                 Exceptions.Throw(new IllegalStateException("Link has been disposed."), Logger);
             }
 
-            string dataType = _reader.ReadString();
-
-            if (dataType == null)
-            {
-                return default(T);
-            }
-
             try
             {
-                T value = (T) _injector.ForkInjector().GetInstance(dataType);
-                value.Read(_reader);
+                T value = _streamingCodec.Read(_reader);
                 return value;
             }
-            catch (InjectionException)
+            catch (Exception e)
```

Comment:
You need to pass the exception to the logger as well. Also, this would be a `WARNING` or `SEVERE`, not merely an `INFO`